tests/run-make/staticlib-broken-bitcode/rmake.rs

```diff
@@ -0,0 +1,23 @@
+// Regression test for https://github.com/rust-lang/rust/issues/128955#issuecomment-2657811196
+// which checks that rustc can read an archive containing LLVM bitcode with a
+// newer version from the one rustc links against.
+use run_make_support::{llvm_ar, path, rfs, rust_lib_name, rustc, static_lib_name};
+
+fn main() {
+    rfs::create_dir("archive");
+
+    // Build a thin archive
+    // FIXME this bitcode file is actually accepted by LLVM.
+    rfs::write("archive/invalid_bitcode.o", b"BC\xC0\xDEaaaa");
```

Comment:
Quoting the LLVM source:

> // We may be consuming bitcode from a client that leaves garbage at the end
> // of the bitcode stream (e.g. Apple's ar tool). If we are close enough to
> // the end that there cannot possibly be another module, stop looking.

https://llvm.org/doxygen/BitcodeReader_8cpp_source.html#l08381

so you may just need more bytes?

Reply:
That indeed worked. Thanks!